rust/lance-table/src/format/fragment.rs

```diff
@@ -163,6 +163,32 @@ impl From<&pb::DeletionFile> for DeletionFile {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BlobFile {
+    /// Path to the blob file (relative to the dataset)
+    pub path: String,
+    /// Number of items in the blob file
+    pub num_items: u32,
```

Comment:
I'm kind of surprised we don't store any offsets. Make it less self contained. One couldn't introspect a blob file to see what was in each item, without having found an offset and length somewhere in a datafile.

Though I guess offsets would be `O(num_rows)` size complexity, so probably impractical to store centrally.